crates/router/src/core/payments/payment_methods.rs

```diff
@@ -0,0 +1,270 @@
+        .list_enabled_connector_accounts_by_profile_id(
+            key_manager_state,
+            profile.get_id(),
+            &key_store,
+            common_enums::ConnectorType::PaymentProcessor,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("error when fetching merchant connector accounts")?;
+
+    let response = PaymentMethodsEnabled::from_payment_connectors_list(payment_connector_accounts)
+        .perform_filtering()
+        .get_required_fields(RequiredFieldsInput::new())
+        .perform_surcharge_calculation()
+        .generate_response();
+
+    Ok(hyperswitch_domain_models::api::ApplicationResponse::Json(
+        response,
+    ))
+}
+
+/// Holds the payment methods enabled for a connector along with the connector name
+/// This struct is a flattened representation of the payment methods enabled for a connector
+#[derive(Debug)]
+struct PaymentMethodsEnabledForConnector {
+    payment_methods_enabled: common_utils::types::RequestPaymentMethodTypes,
+    payment_method: common_enums::PaymentMethod,
+    connector: String,
+}
+
+/// Container for the inputs required for the required fields
+struct RequiredFieldsInput {}
+
+impl RequiredFieldsInput {
+    fn new() -> Self {
+        Self {}
+    }
+}
+
+/// Container for the filtered payment methods
+struct FilteredPaymentMethodsEnabled {
+    payment_methods_enabled: Vec<PaymentMethodsEnabledForConnector>,
+}
+
+impl FilteredPaymentMethodsEnabled {
+    fn get_required_fields(
+        self,
+        _input: RequiredFieldsInput,
+    ) -> PaymentMethodsEnabledWithRequiredFieldsContainer {
+        let required_fields_info = self
+            .payment_methods_enabled
+            .into_iter()
+            .map(
+                |payment_methods_enabled| PaymentMethodsEnabledWithRequiredFields {
+                    required_field: None,
+                    payment_method_subtype: payment_methods_enabled
+                        .payment_methods_enabled
+                        .payment_method_type,
+                },
+            )
+            .collect();
+
+        PaymentMethodsEnabledWithRequiredFieldsContainer {
+            payment_methods_enabled: required_fields_info,
+        }
+    }
+}
+
+/// Element container to hold the filtered payment methods with required fields
+struct PaymentMethodsEnabledWithRequiredFields {
+    required_field:
+        Option<std::collections::HashMap<String, api_models::payment_methods::RequiredFieldInfo>>,
+    payment_method_subtype: common_enums::PaymentMethodType,
+}
+
+/// Container to hold the filtered payment methods enabled with required fields
+struct PaymentMethodsEnabledWithRequiredFieldsContainer {
+    payment_methods_enabled: Vec<PaymentMethodsEnabledWithRequiredFields>,
+}
+
+/// Element Container to hold the filtered payment methods enabled with required fields and surcharge
+struct PaymentMethodsEnabledWithRequiredFieldsAndSurcharge {
```

Comment:
we can rename `RequiredFieldsAndSurchargeForEnabledPaymentMethodSubType`, so that the `Vec<RequiredFieldsAndSurchargeForEnabledPaymentMethodSubType>` can be RequiredFieldsAndSurchargeForEnabledPaymentMethodSubTypes